### PR TITLE
Remove a big chunk of jQuery, use module syntax

### PR DESCRIPTION
**Pull request by RobbieTheWagner**
Removed a bunch of jQuery and refactored to import ember modules. Still a work in progress.

### CODE REVIEW

**Comment by RobbieTheWagner** [comment]
@YoranBrondsema if you wouldn't mind giving this a review, that would be great!

File: README.md [hunk 1]
```diff
@@ -23,14 +23,13 @@ Install the addon.
 Import the mixins in a component and use the events or the handlers.
 
 ```javascript
-import Ember from 'ember';
+import Component from '@ember/component';
```
Comment (YoranBrondsema) [inline README.md:26]: From what version of Ember is this way of importing modules supported? I see from Travis that it's at least 2.8 but maybe the earliest supported version should be mentioned somewhere in the README.

File: tests/helpers/start-app.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
-import Ember from 'ember';
 import Application from '../../app';
 import config from '../../config/environment';
+import { merge } from '@ember/polyfills';
+import { run } from '@ember/runloop';
 
 export default function startApp(attrs) {
-  let attributes = Ember.merge({}, config.APP);
-  attributes = Ember.merge(attributes, attrs); // use defaults, but you can override;
+  let attributes = merge({}, config.APP);
```
Comment (YoranBrondsema) [inline tests/helpers/start-app.js:7]: I think `merge` is deprecated in favor of `assign` (see https://github.com/emberjs/ember.js/pull/12303).

File: tests/integration/components/respond-to-resize-test.js [hunk 3]
```diff
@@ -1,43 +1,47 @@
...
-import Ember from 'ember';
 import hbs from 'htmlbars-inline-precompile';
+import { triggerEvent } from 'ember-native-dom-helpers';
 
-moduleForComponent('respond-to-resize', {
+moduleForComponent('respond-to-resize', 'Integration | Component | respond to resize', {
   integration: true,
-  beforeEach: function () {
+  beforeEach: function() {
     window.requestAnimationFrame = setTimeout;
```
Comment (YoranBrondsema) [inline tests/integration/components/respond-to-resize-test.js]: I don't think this line is necessary anymore as the tests on CI are run on Chrome and not on PhantomJS. So `window.requestAnimationFrame` is correctly defined.

File: tests/integration/components/respond-to-scroll-test.js [hunk 4]
```diff
@@ -1,32 +1,32 @@
...
-import Ember from 'ember';
 import hbs from 'htmlbars-inline-precompile';
+import { triggerEvent } from 'ember-native-dom-helpers';
 
-moduleForComponent('respond-to-scroll', {
+moduleForComponent('respond-to-scroll', 'Integration | Component | respond to scroll', {
   integration: true,
-  beforeEach: function () {
+  beforeEach: function() {
     window.requestAnimationFrame = setTimeout;
```
Comment (YoranBrondsema) [inline tests/integration/components/respond-to-scroll-test.js]: Same

**Comment by YoranBrondsema** [comment]
@rwwagner90 Great changes, thanks! Just left a few small comments.

File: tests/helpers/start-app.js (hunk 2 above)
Comment (RobbieTheWagner) [reply tests/helpers/start-app.js:7]: Merge is still used in the blueprints. Likely for compatibility with all ember versions.

File: README.md (hunk 1 above)
Comment (RobbieTheWagner) [reply README.md:26]: This is supported by ember-cli-babel 6.6.0+. We have that as a dependency of the addon, so it works.

File: tests/integration/components/respond-to-resize-test.js (hunk 3 above)
Comment (RobbieTheWagner) [reply tests/integration/components/respond-to-resize-test.js]: 👍 fixed

**Comment by RobbieTheWagner** [comment]
@YoranBrondsema responded to your comments, thanks!

File: README.md (hunk 1 above)
Comment (YoranBrondsema) [reply README.md:26]: Ah OK, I thought it was dependent on the version of Ember.js. Cool then!

File: tests/helpers/start-app.js (hunk 2 above)
Comment (YoranBrondsema) [reply tests/helpers/start-app.js:7]: :+1:

**Comment by YoranBrondsema** [comment]
LGTM. Thanks for this PR and the previous clean up. The code was so 2016 :-).

Since you have the permissions and it's your PR, I'll let you merge! Then we should probably cut out a new release.

**Comment by RobbieTheWagner** [comment]
@YoranBrondsema I think there were a couple leftover jquery things I did not remove yet. Might want to wait to cut a release until I fully remove it. Specifically, do you know why this is outside of the mixin? https://github.com/dollarshaveclub/ember-responds-to/blob/master/addon/mixins/responds-to-esc-keydown.js#L9-L20

I would think we should put it in an `init` or `didInsertElement` or something, rather than floating outside.

**Comment by YoranBrondsema** [comment]
@rwwagner90 Of course, it makes sense to wait then.

The original commit that introduced the code was by @pwfisher, although I'm not 100% sure he wrote it. To me it seems to be that it's an optimization to reduce the number of event listeners: we only have 1 listener instead of one per mixin consumer. But then the browser is also maintaining a list of listeners for each event, which functionality we're basically reimplementing here. So maybe it would be interesting to benchmark and compare? I.e. measure this vs setting up the listener in `didInsertElement` (as is done in `responds-to-resize` for instance) for a couple of thousands component instances. This way we would know for sure which way is best :-). I wouldn't mind doing that since you've already done a lot on the addon lately, but of course, if you have time and willpower, feel free!

That aside, I still think it's possible to de-jQuery this piece of code (along with responds-to-enter-keydown) without altering its behavior.

**Pull request merged**